keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactVersionLinks.kt

```diff
@@ -51,22 +52,40 @@ class ArtifactVersionLinks(private val scmInfo: ScmInfo, private val cacheFactor
     } else null
   }
 
+  fun getScmType(commitLink: String): String? {
+    val commitURL = URL(commitLink)
+
+    val scmBaseURLs = runBlocking {
+      val cachedValue = cache[cacheName].get()
+      cachedValue ?: scmInfo.getScmInfo()
+    }
+
+    val base = scmBaseURLs.filter { (_,baseUrl) ->
+      commitURL.host == URL(baseUrl).host
+    }
+
+    return base.keys.toList()[0]
```

Comment:
This could throw index out of boundary exception. If base is not found after filter, we should return null and log.

Reply:
didn't know about firstOrNull!, but I did use getOrNull let me know if you want me to change it.